README.md
## Switch React Native

<p align="center">
    <img src ="https://media.giphy.com/media/EP7FFmBKLhfq3n8pLo/giphy.gif" />
</p>

### Content
- [Switch React Native](#switch-react-native)
  - [Content](#content)
  - [Installation](#installation)
  - [Usage example](#usage-example)
  - [Properties](#properties)
  - [Events](#events)
  - [License](#license)
  - [Questions?](#questions)

### Installation
```bash
npm install switch-react-native
```
### Usage example
```javascript
import React, { Component } from 'react';
import { View } from 'react-native';
import { Switch } from 'switch-react-native';

class SwitchExample extends Component {
  render() {
    return (
      <View>
        <Switch
          height={40}
          width={300}
          activeText={`Active Text`}
          inActiveText={`InActive Text`}
          onValueChange={(value: any) => console.log(value)}
        />
      </View>
    );
  }
}
```
### Properties
* `width` (Number) - Initial switch state (default: 150),
* `height` (Number) - Initial switch state (default: 50),
* `time` (Number) - Initial switch state (default: 200 mms),
* `value` (Bool) - Initial switch state (default: false),
* `activeText` (String) - String activeText,
* `inActiveText` (String) - String inActiveText,
* `activeTextStyle` (Object) - Style is object,
* `inActiveTextStyle` (Object) - Style is object,
* `activeStyle` (Object) - Style is object,
* `backgroundStyle` (Object) - Style is object,

### Events
* `onValueChange`: This function is called when the switch is activated.

### License
MIT License

### Questions?
Feel free to [create an issue](https://github.com/jundat95/switch-react-native/issues)


CHANGELOG.md
# Changelog

Switch, Tabbar support React Native


## [v1.0.8](https://github.com/jundat95/switch-react-native/releases/tag/v1.0.8) (2018/10/29)

**Implemented enhancements:**

- Update lib support react native typescript

**Fixed bug:**


**Merged pull requests:**

lib/switch.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Animated, Easing,
  TouchableOpacity,
  View, Text,
} from 'react-native';

export class Switch extends Component {

  static propTypes = {
    width: PropTypes.number,
    height: PropTypes.number,
    time: PropTypes.number,
    value: PropTypes.bool,
    onValueChange: PropTypes.func,
    activeText: PropTypes.string,
    inActiveText: PropTypes.string,
    textStyle: PropTypes.object,
    activeStyle: PropTypes.object,
    backgroundStyle: PropTypes.object,
  };

  static defaultProps = {
    width: 150,
    height: 50,
    time: 500,
    value: false,
  };

  constructor(props, context) {
    super(props, context);

    this.state = {
      value: this.props.value,
      marginLeftAnim: new Animated.Value(0),
      offsetWidth: this.props.width / 2,
      offsetHeight: this.props.height,
      activeText: this.props.activeText,
      inActiveText: this.props.inActiveText,
    };
  }

  _animateSwitch = (value) => {
    Animated.timing(
      this.state.marginLeftAnim,
      {
        toValue: value ? 0 : this.state.offsetWidth,
        duration: this.props.time,
        easing: Easing.bezier(0.68, -0.55, 0.265, 1.55),
      }
    ).start();
    this.setState({ value: value });
  };

  render() {
    const { marginLeftAnim } = this.state;
    const { textStyle, activeStyle, backgroundStyle } = this.props;

    return (
      <View
        style={[
          styles.container,
          {
            width: this.props.width,
            height: this.props.height,
          },
          backgroundStyle,
        ]}>
        <TouchableOpacity
          onPressIn={this._pressButtonLeft}
          style={[styles.button, {
            width: this.state.offsetWidth,
            height: this.state.offsetHeight,
          }]}>
          <Text style={[ styles.text, textStyle]}>{this.state.activeText}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPressIn={this._pressButtonRight}
          style={[styles.button, {
            width: this.state.offsetWidth,
            height: this.state.offsetHeight,
          }]}>
          <Text style={[styles.text, textStyle]}>{this.state.inActiveText}</Text>
        </TouchableOpacity>
        <Animated.View
          style={[
            {
              marginLeft: marginLeftAnim,
              width: this.state.offsetWidth,
              height: this.state.offsetHeight,

            },
            styles.active,
            activeStyle,
          ]}
        />
      </View>
    );

  }

  _pressButtonLeft = () => {
    this._animateSwitch(true);
    this.props.onValueChange(true);
  };

  _pressButtonRight = () => {
    this._animateSwitch(false);
    this.props.onValueChange(false);
  };

}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#ffffff',
    flexDirection: 'row',
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowRadius: 2,
    shadowOpacity: 0.2,
    elevation: 5,
    position: 'relative',
    zIndex: 1,
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    zIndex: 1,
  },
  text: {
    fontWeight: 'bold',
    fontSize: 14,
    color:  '#215e79',
  },
  active: {
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#dae6e8',
  },
});


components/switch-component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const react_native_1 = require("react-native");
class Switch extends React.Component {
    constructor(props) {
        super(props);
        this._animateSwitch = (value) => {
            react_native_1.Animated.timing(this.state.marginLeftAnim, {
                toValue: value ? 0 : this.state.offsetWidth,
                duration: this.props.time,
                easing: react_native_1.Easing.bezier(0.68, -0.55, 0.265, 1.55),
            }).start();
            this.setState({ value: value });
            console.log(this.state.value);
        };
        this._pressButtonLeft = () => {
            this._animateSwitch(true);
            this.props.onValueChange(true);
        };
        this._pressButtonRight = () => {
            this._animateSwitch(false);
            this.props.onValueChange(false);
        };
        this.state = {
            value: this.props.value ? this.props.value : true,
            marginLeftAnim: new react_native_1.Animated.Value(0),
            offsetWidth: this.props.width / 2,
            offsetHeight: this.props.height,
            activeText: this.props.activeText ? this.props.activeText : 'InActive Text',
            inActiveText: this.props.inActiveText ? this.props.inActiveText : 'Active Text',
        };
    }
    render() {
        let { marginLeftAnim } = this.state;
        let { activeStyle, backgroundStyle, activeTextStyle, inActiveTextStyle } = this.props;
        return (<react_native_1.View style={[
            styles.container,
            {
                width: this.props.width,
                height: this.props.height,
            },
            backgroundStyle,
        ]}>
        <react_native_1.TouchableOpacity onPressIn={this._pressButtonLeft} style={[styles.button, {
                width: this.state.offsetWidth,
                height: this.state.offsetHeight,
            }]}>
          {this.state.value ?
            <react_native_1.Text style={[styles.textActive, activeTextStyle]}>{this.state.activeText}</react_native_1.Text>
            :
                <react_native_1.Text style={[styles.textInActive, inActiveTextStyle]}>{this.state.activeText}</react_native_1.Text>}
        </react_native_1.TouchableOpacity>
        <react_native_1.TouchableOpacity onPressIn={this._pressButtonRight} style={[styles.button, {
                width: this.state.offsetWidth,
                height: this.state.offsetHeight,
            }]}>
          {this.state.value ?
            <react_native_1.Text style={[styles.textInActive, inActiveTextStyle]}>{this.state.inActiveText}</react_native_1.Text>
            :
                <react_native_1.Text style={[styles.textActive, activeTextStyle]}>{this.state.inActiveText}</react_native_1.Text>}
        </react_native_1.TouchableOpacity>
        <react_native_1.Animated.View style={[
            {
                marginLeft: marginLeftAnim,
                width: this.state.offsetWidth,
                height: this.state.offsetHeight,
            },
            styles.active,
            activeStyle,
        ]}/>
      </react_native_1.View>);
    }
}
exports.default = Switch;
const styles = react_native_1.StyleSheet.create({
    container: {
        backgroundColor: '#ffffff',
        flexDirection: 'row',
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowRadius: 2,
        shadowOpacity: 0.2,
        elevation: 5,
        position: 'relative',
        zIndex: 1,
    },
    button: {
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'transparent',
        zIndex: 1,
    },
    textActive: {
        fontWeight: 'bold',
        fontSize: 14,
        color: '#215e79',
    },
    textInActive: {
        fontWeight: 'bold',
        fontSize: 14,
        color: '#215e79',
    },
    active: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#dae6e8',
    },
});
